lib/actions/apiV2.js

```diff
@@ -920,11 +923,28 @@ export function routingQuery(searchId = null, updateSearchInReducer) {
           routingError,
           {
             rewritePayload: (response, dispatch, getState) => {
-              const withCollapsedShortNames =
-                response.data?.plan?.itineraries?.map((itin) => ({
+              const itineraries = response.data?.plan?.itineraries
+              const activeModeStrings = activeModes.map(
+                (am) => SIMPLIFICATIONS[am.mode]
+              )
+              let filteredItineraries
+
+              if (strictModes) {
+                filteredItineraries = itineraries.filter((itin) =>
+                  itin.legs.some((leg) =>
+                    activeModeStrings.includes(SIMPLIFICATIONS[leg.mode])
+                  )
+                )
+              } else {
+                filteredItineraries = itineraries
+              }
+
+              const withCollapsedShortNames = filteredItineraries.map(
+                (itin) => ({
```

Comment:
What are we doing with this new variable?

Reply:
withCollapsedShortNames isn't new

Reply:
Ah might be a weird diff then. Either way please comment!

Reply:
This and example config updated in [15b71fc](https://github.com/opentripplanner/otp-react-redux/pull/1024/commits/15b71fc61007ac7ecc59175566d53e96f758d3cb)!